src/query/procedure/module.cpp

```diff
@@ -832,10 +832,28 @@ PythonModule::~PythonModule() {
 }
 
 bool PythonModule::Load(const std::filesystem::path &file_path) {
+  // This configures python to load share libs with RTLD_NOW and RTLD_DEEPBIND
+  // flags, which in turn solves the issue of loading wrong libstdc++ lib.
+  static const auto initialize = std::invoke([]() {
+    auto gil = py::EnsureGIL();
+    auto *flag = PyLong_FromLong(RTLD_NOW | RTLD_DEEPBIND);
+    auto *setdl = PySys_GetObject("setdlopenflags");
+    MG_ASSERT(setdl);
+    auto *arg = PyTuple_New(1);
+    MG_ASSERT(arg);
+    MG_ASSERT(PyTuple_SetItem(arg, 0, flag) == 0);
+    PyObject_CallObject(setdl, arg);
+    Py_DECREF(flag);
+    Py_DECREF(setdl);
+    Py_DECREF(arg);
+    return true;
+  });
```

Comment:
This cannot be moved in the constructor since the `generator.py` script runs memgraph binary during build time and it fails to run it, if this is set in the ctor